ReactAndroid/src/main/java/com/facebook/react/modules/blob/BlobModule.java

```diff
@@ -0,0 +1,145 @@
+
+import android.content.ContentResolver;
+import android.content.res.Resources;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.annotation.Nullable;
+import android.webkit.MimeTypeMap;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.module.annotations.ReactModule;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@ReactModule(name = "BlobModule")
+public class BlobModule extends ReactContextBaseJavaModule {
+
+
+  public BlobModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "BlobModule";
+  }
+
+  @Override
+  @Nullable
+  public Map getConstants() {
+    // The application can register BlobProvider as a ContentProvider so that blobs are resolvable.
+    // If it does, it needs to tell us what authority was used via this string resource.
+    Resources resources = getReactApplicationContext().getResources();
+    String packageName = getReactApplicationContext().getPackageName();
+    int resourceId = resources.getIdentifier("blob_provider_authority", "string", packageName);
+    if (resourceId == 0) {
+      return null;
+    }
+
+    return MapBuilder.of(
+      "BLOB_URI_SCHEME", "content",
+      "BLOB_URI_HOST", resources.getString(resourceId));
+  }
+
+  private static Map<String, byte[]> sBlobs = new HashMap<>();
+
+  public static String store(byte[] data) {
+    String blobId = UUID.randomUUID().toString();
+    store(data, blobId);
+    return blobId;
+  }
+
+  public static void store(byte[] data, String blobId) {
+    sBlobs.put(blobId, data);
+  }
+
+  public static void remove(String blobId) {
+    sBlobs.remove(blobId);
+  }
+
+  @Nullable
+  public static byte[] resolve(Uri uri) {
+    String blobId = uri.getLastPathSegment();
+    int offset = 0;
+    int size = -1;
+    String offsetParam = uri.getQueryParameter("offset");
+    if (offsetParam != null) {
+      offset = Integer.parseInt(offsetParam, 10);
+    }
+    String sizeParam = uri.getQueryParameter("size");
+    if (sizeParam != null) {
+      size = Integer.parseInt(sizeParam, 10);
+    }
+    return resolve(blobId, offset, size);
+  }
+
+  @Nullable
+  public static byte[] resolve(String blobId, int offset, int size) {
+    byte[] data = sBlobs.remove(blobId);
```

Comment:
Why remove it? From the name of this method it looks like it should just return the data.